### PR TITLE
feat: Use wadm-client and wadm-types crates for wadm interactions

### PR DESCRIPTION
**Pull request by joonas**
## Feature or Problem

Switching us from using `wash-lib` to the newly available `wadm-client` and `wadm-types` crates instead, which cleans things up nicely!

cc @thomastaylor312

### CODE REVIEW

File: src/services.rs [hunk 1]
```diff
@@ -275,26 +272,20 @@ impl ServiceWatcher {
     /// This intended to be called by the controller whenever it reconciles state.
     pub async fn reconcile_services(&self, apps: Vec<ModelSummary>, lattice_id: String) {
         if let Some(watcher) = self.watchers.read().await.get(lattice_id.as_str()) {
+            let wadm_client =
+                WadmClient::from_nats_client(&lattice_id, None, watcher.nats_client.clone());
```
Comment (joonas) [inline src/services.rs:276]: I don't necessarily love this, having to instantiate a new `wadm_client::Client` each time we reconcile, maybe it should be possible to do something like `.set_lattice(...)` with the `wadm_client::Client` to switch between them, since all that currently does is generate different subjects for the various endpoints

Comment (thomastaylor312) [reply src/services.rs:276]: From a code perspective, there really isn't a ton of overhead, but if there is an improvement to the client you need, let me know

File: src/resources/application.rs [hunk 2]
```diff
@@ -648,34 +568,24 @@ pub async fn patch_application(
...
-            .await
-        {
-            Ok(res) => res,
-            Err(e) => {
-                return internal_error(anyhow!("unable to find app: {}", e));
+    let wadm_client = WadmClient::from_nats_client(&lattice_id, None, nats_client);
+    let current_manifest = match wadm_client.get_manifest(&name, None).await {
+        Ok(m) => m,
+        Err(e) => match e {
+            ClientError::ManifestLoad(_)
```
Comment (thomastaylor312) [inline src/resources/application.rs:575]: Nit: You could just shorten this to `_ => return internal_error(anyhow!("unable to request app from wadm: {}", e))` after the `ClientError::NotFound` branch

File: src/resources/application.rs [hunk 3]
```diff
@@ -684,52 +594,25 @@ pub async fn patch_application(
...
-        _ => {
-            // For now we have to check the error message to see if we can continue,
-            // despite getting an error.
-            if !put.message.contains("already exists") {
-                return internal_error(anyhow!(
-                    "unexpected response from wadm: result={:?}, message={}",
-                    put.result,
-                    put.message
-                ));
+            ClientError::ManifestLoad(_)
```
Comment (thomastaylor312) [inline src/resources/application.rs:608]: Same note here around making this match cleaner

File: src/resources/application.rs [hunk 4]
```diff
@@ -759,67 +642,27 @@ pub async fn delete_application(
...
-            "unexpected response from wadm: result={:?}, message={}",
-            delete.result,
-            delete.message
-        )),
+    let wadm_client = WadmClient::from_nats_client(&lattice_id, None, nats_client);
+    match wadm_client.delete_manifest(&name, None).await {
+        Ok(_) => Json(Application::new(name)).into_response(),
+        Err(e) => match e {
+            ClientError::NotFound(_) => not_found_error(anyhow!("apps \"{}\" not found", name)),
+            ClientError::Other(_)
```
Comment (thomastaylor312) [inline src/resources/application.rs:650]: Same note here about cleaner match

**Review by thomastaylor312 — APPROVED** [review]
Nothin' but nits

File: src/resources/application.rs (hunk 2 above)
Comment (joonas) [reply src/resources/application.rs:575]: Good call, that would read much nicer, thanks for the reminder!

File: src/services.rs (hunk 1 above)
Comment (joonas) [reply src/services.rs:276]: Agreed, I thinking it would be nice to store a `WadmClient` instance instead of the `async_nats::Client` in the  `Watcher` since it isn't interacting with anything other than wadm in this case, which having the ability to swap between lattices would enable.

I'm not pressed either way, it felt like not having to recreate the WadmClient every time you want to operate on a different lattice could be a nice improvement from usability perspective
